config/management/genesis/src/fullnode_builder.rs

```diff
@@ -84,6 +109,58 @@ impl FullnodeConfig {
         }
     }
 
+    fn attach_to_validator(&mut self, validator_config: &mut NodeConfig) -> Result<()> {
+        ensure!(
+            matches!(validator_config.base.role, RoleType::Validator),
+            "Validator config must be a Validator config"
+        );
+
+        // Grab the public network config from the validator and insert it into the VFN's config
+        let public_network = {
+            let (i, _) = validator_config
+                .full_node_networks
+                .iter()
+                .enumerate()
+                .find(|(_i, config)| config.network_id == NetworkId::Public)
+                .expect("Validator should have a public network");
+            validator_config.full_node_networks.remove(i)
```

Comment:
Why do you remove it instead of clone it?  Oh, I see you transfer the public network from the validator to the VFN.

Reply:
Yep :) that way both the validator and the VFN aren't fighting over the port